### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

**Comment by TrySound** [comment]
On my mac all tests have been passed.

**Comment by TrySound** [comment]
Maybe it's time to stop thinking about 0.8?

**Comment by thlorenz** [comment]
What's the big gain here? Are things not working with the current versions?
Seems like newer ones break 0.8?

**Comment by TrySound** [comment]
@thlorenz On my mac tests passes. But travis fails somehow.

**Comment by es128** [comment]
Looks like it might merely be an issue of the old npm bundled with node 0.8.x not supporting the caret semver operator, not that compatibility was actually broken. There's a fix by adding something like `npm install -g npm@2` to the travis script for the 0.8 environment. Will try to dig up an example.

**Comment by es128** [comment]
Try adding

```yml
before_install:
  -  if [ "$TRAVIS_NODE_VERSION" == "0.8" ]; then npm update -g npm; fi
```

**Comment by TrySound** [comment]
@es128 Woohoo! It works!

**Comment by TrySound** [comment]
@thlorenz Patch release?

File: package.json [hunk 1]
```diff
@@ -33,13 +33,13 @@
   },
   "dependencies": {
     "graceful-fs": "~4.1.2",
-    "minimatch": "~0.2.12",
-    "readable-stream": "~1.0.26-2"
+    "minimatch": "^2.0.10",
```
Comment (thlorenz) [inline package.json]: Pleas no carets, I don't trust minor versions _yet_.
So updating the versions is fine but keep the `~` range specifier. for all of these.
To have `npm install --save` do that for you do `npm config set save-prefix='~'`.

This applies to all dependencies including dev-dependencies.

File: .travis.yml [hunk 2]
```diff
@@ -4,3 +4,5 @@ node_js:
   - "0.10"
   - "0.12"
   - "iojs-v2.4"
+before_install:
+  - if [ "$TRAVIS_NODE_VERSION" == "0.8" ]; then npm update -g npm; fi
```
Comment (thlorenz) [inline .travis.yml]: This is no longer needed once you fix all the `^` problems as I indicated below.

**Comment by thlorenz** [comment]
I'd still like you to answer my question: 
> What's the big gain here? Are things not working with the current versions?

Maybe @es128 can help with an answer here?

Also after you fix all issues I outlined please squash into one commit.

@es128 could you please merge this into master after all has been addressed and we established that this PR actually improves things? I'll publish a minor upgrade once that's done.

**Comment by TrySound** [comment]
@thlorenz The main reason is that need to download too much packages with different versions. Even npm3 with dedupe can't fix that.

**Comment by thlorenz** [comment]
Thanks, I guess that's reason enough as long as nothing breaks. Please squash and @es128 feel free to merge if you got time (otherwise I'll try to find some time).

**Comment by TrySound** [comment]
Heh! Maybe this fix still needed for dependencies of dependencies :)

**Comment by thlorenz** [comment]
@TrySound good point, you are correct. Please leave that fix in -- my bad.

**Comment by TrySound** [comment]
@thlorenz It's ok. Just an experience for me.

**Comment by es128** [comment]
> The main reason is that need to download too much packages with different versions. Even npm3 with dedupe can't fix that.

The thing about it is that this is going to be a constantly moving target, particularly because of sticking with the tilde semver operator. I don't object to updating the deps and think its quite safe (that's what the unit tests and travis are for), but that doesn't change the fact that it's a pretty minor win that will be short-lived anyway unless we go through this exercise on a regular basis.

I think there's a choice to be made here between modernizing to the state of most node modules out there with the caret (largely in response to npm@3's flattening), or being one of those modules that tries to stay more frozen on a stable state in order to avoid maintenance hassles that occur when something else in the ecosystem interferes.

**Comment by TrySound** [comment]
@es128 Maybe somebody will go along all used packages and update them.

**Comment by thlorenz** [comment]
>  in order to avoid maintenance hassles

Actually it's a bit more of a hassle since updates need to happen more often.
On the other hand it avoids tons of issues caused by deps incompatibilities that take time to triage, rather I control deps more tightly.

So here is the deal. I'm open to try switching to the caret for readdirp if you guys are willing to help me with answering/closing issues caused by these incompatibilities.
If you agree I'll make you collab @TrySound so you can do that.

If my worries are pointless no more issues will come up, but otherwise I'm at least not alone with them :)

**Comment by es128** [comment]
I'm actually agnostic on the deps strategy, but either way I'll certainly be here to help if there are ever any issues.

**Comment by TrySound** [comment]
@thlorenz Sure, I will help :)

**Comment by TrySound** [comment]
@thlorenz ping :)

**Comment by thlorenz** [comment]
pong :)
As I said since I have your support I'm ok with this.
Please merge into master and I'll publish a new minor.

**Comment by es128** [comment]
Landed as d5c5c65060a09cb3ddf5c184a6e26422210ef784

**Comment by thlorenz** [comment]
Thanks published as 2.0

**Comment by TrySound** [comment]
@thlorenz Why? It's just a patch. No new API, no breaking changes. For now every dependent package should update dep version. Semver is good thing. Need to follow it.

**Comment by thlorenz** [comment]
Nope we changed lots of versions of the deps and additionally the entire versioning scheme of them.

**Pull request closed without merging**